tapdb/sqlc/migrations/000002_assets.up.sql

```diff
@@ -143,7 +143,19 @@ CREATE TABLE IF NOT EXISTS managed_utxos (
     -- TODO(roasbeef): can then reconstruct on start up to ensure matches up
     merkle_root BLOB NOT NULL CHECK(length(merkle_root) = 32),
 
-    txn_id INTEGER NOT NULL REFERENCES chain_txns(txn_id)
+    txn_id INTEGER NOT NULL REFERENCES chain_txns(txn_id),
+    
+    -- The identity of the application that currently has a lease on this UTXO.
+    -- If NULL, then the UTXO is not currently leased. A lease means that the
+    -- UTXO is being reserved/locked to be spent in an upcoming transaction and
+    -- that it should not be available for coin selection through any of the
+    -- wallet RPCs.
+    lease_owner BLOB,
```

Comment:
Could alternatively be a distinct table, then the unlock/lock process is an insert/delete vs modification.

Reply:
I did look into that but found that it makes the coin selection SELECT statement quite a bit more complicated. Because then we'd need to do a `LEFT JOIN` on that leases table and also have WHERE statements that evaluates those columns based on the `leased` parameter.

Reply:
Hmm not sure I follow, for lease aware coin selection, don't we just need to first get the set of leased asset IDs (the internal primary key one), then assert that the returned set isn't found amongst that table? So imagining it as a CTE. 

In either case will check out the latest posted version here.